### PR TITLE
Helper functions for AOT Autograd testing

### PR DESCRIPTION
**Pull request by anijain2305**
More helper functions for AOT Autograd testing

### CODE REVIEW

File: torchdynamo/testing.py [hunk 1]
```diff
@@ -21,6 +21,41 @@
...
+    ):
+        return reduce_out(out.logits)
+    elif type(out).__name__ == "SquashedNormal":
+        return out.mean
+    elif isinstance(out, dict):
+        return sum([reduce_out(value) for value in out.values()])
+    raise NotImplementedError("Don't know how to reduce")
+
+
+def checkpoint_module_params(mod):
```
Comment (jansel) [inline torchdynamo/testing.py:43]: Could this be combined with https://github.com/jansel/torchdynamo/blob/main/torchdynamo/utils.py#L292 ?

File: torchdynamo/testing.py [hunk 2]
```diff
@@ -21,6 +21,41 @@
...
+    with torch.no_grad():
+        rng_state = torch.clone(torch.random.get_rng_state())
+        module_state = {}
+        for k, v in mod.state_dict().items():
+            module_state[k] = torch.clone(v)
+
+    def restore():
+        with torch.no_grad():
+            torch.random.set_rng_state(rng_state)
+            mod.load_state_dict(module_state)
```
Comment (jansel) [inline torchdynamo/testing.py:54]: Is this correct if there is aliasing?

**Review by jansel — CHANGES_REQUESTED** [review]

File: torchdynamo/testing.py (hunk 2 above)
Comment (anijain2305) [reply torchdynamo/testing.py:54]: Hmm, I did not consider aliasing in the parameters itself.
I think I will just use your checkpoint_params from utils.py.

**Comment by anijain2305** [comment]
Changes done.

**Review by jansel — APPROVED** [review]

**Pull request merged**